### PR TITLE
Add unit tests to several API utility functions

### PR DESCRIPTION
**Pull request by azrikahar**
## Description

Adds unit test for some utility functions in the API.

## Type of Change

- [ ] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [x] Other, please describe: Tests

## Requirements Checklist

- [x] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

**Review by br41nslug — COMMENTED** [review]
Yes more tests! 🚀 Locally for me 2 tests in `get-date-formatted.test` are failing unfortunately is this a configuration error on my side?
image

**Comment by azrikahar** [comment]
> Yes more tests! 🚀 Locally for me 2 tests in `get-date-formatted.test` are failing unfortunately is this a configuration error on my side? 

I was super curious why the first and fourth tests passed for you as they aren't too different (in terms of format):

https://github.com/directus/directus/blob/e1e271f93bd8521a1cf234a474469d9dfb2797c4/api/src/utils/get-date-formatted.test.ts#L14-L17

but thankfully the screenshot you shared which showed it is off by 1 hour was a great indicator that this might be a daylight saving issue 🤯 

I was trying to add more variety with 2nd test being June and 3rd test being October, but the `getTimezoneOffset()` for those 2 dates is actually different than the first (January) and last (November) test cases as described in: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset#gettimezoneoffset_and_dst

TIL! I've updated all of them to use January and November for consistency since all are using `1`s now, especially since the main thing to test is single/double digit month or day, and not the actual month itself. Can you try again to see does it work on your machine?

File: api/src/utils/get-date-formatted.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	return nowUTC;
+}
+
+// currently only uses January and November dates to account for daylight saving
+test.each([
+	{ utc: '2023-01-01T01:23:45.678Z', expected: '20230101-12345' },
+	{ utc: '2023-01-11T01:23:45.678Z', expected: '20230111-12345' },
+	{ utc: '2023-11-01T01:23:45.678Z', expected: '20231101-12345' },
+	{ utc: '2023-11-11T12:34:56.789Z', expected: '20231111-123456' },
```
Comment (br41nslug) [inline api/src/utils/get-date-formatted.test.ts:29]:
```suggestion
	{ utc: '2023-11-01T01:23:45.678Z', expected: '20231101-12345' },
	{ utc: '2023-11-11T12:34:56.789Z', expected: '20231111-123456' },
	{ utc: '2023-06-01T01:23:45.678Z', expected: '20230601-12345' },
	{ utc: '2023-06-11T12:34:56.789Z', expected: '20230611-123456' },
```
Just to be on the safe side testing a summer-time (mar-nov) date would be good to include

**Review by br41nslug — COMMENTED** [review]
can confirm the tests are passing now 😄

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**